docs/source/en/generation_strategies.md

```diff
@@ -465,30 +484,35 @@ If the main and assistant models have different tokenizers, use Universal Assist
 ['Alice and Bob are sitting in a bar. Alice is drinking a beer and Bob is drinking a']
 ```
 
-When using assisted decoding with sampling methods, you can use the `temperature` argument to control the randomness,
-just like in multinomial sampling. However, in assisted decoding, reducing the temperature may help improve the latency.
+#### Prompt Lookup
+
+Alternatively, you can also set the `prompt_lookup_num_tokens` to trigger n-gram based assisted decoding, as opposed
+to model based assisted decoding. You can read more about it [here](https://twitter.com/joao_gante/status/1747322413006643259).
+
+#### Self-Speculative Decoding
+
+An LLM can be trained to also use its language modeling head with earlier hidden states as input, effectivelly
+skipping layers to yield a lower-quality output -- a technique called early exits.
```

Comment:
```suggestion
skipping layers to yield a lower-quality output -- a technique called early exiting.
```

It'd be nice to clarify that this lower-quality output is the output from the self-speculative assistant, but the final generation will be the same as if no assistance had been used.

Reply:
Adding a couple of lines as suggested by @pcuenca 

```suggestion
skipping layers to yield a lower-quality output -- a technique called early exiting.
We use the lower-qulity early exit output as an assistant output, and apply self-speculation to fix the output using the remaining layers. The final generation of that self-speculative solution is the same (or has the same distribution) as the original model's generation.
```